### PR TITLE
packaging/opensuse: fix for /usr/libexec on TW, do not hardcode AppArmor profile path

### PR DESCRIPTION
**Pull request by bboozzoo**
opensuse Tumbleweed has switched to using /usr/libexec, while 15.x stayed with
/usr/lib. Update the RPM spec so that we try to install the right profile.

See https://lists.opensuse.org/opensuse-factory/2019-10/msg00204.html for more details.

@Conan-Kudo can you take a look at the macro bits?

### CODE REVIEW

**Comment by bboozzoo** [comment]
Note, this does not solve the whole of a problem, we have code in dirs/dirs.go that assumes /usr/lib as libexec on openSUSE for all versions.

File: packaging/opensuse/snapd.spec [hunk 1]
```diff
@@ -14,6 +14,9 @@
 
 # Please submit bugfixes or comments via http://bugs.opensuse.org/
 
+# takes an absolute path with slashes and turns it into an AppArmor profile path
+%define as_apparmor_path() %(echo "%1" | tr / . | cut -c2-)
```
Comment (zyga) [inline packaging/opensuse/snapd.spec:18]: I didn't know RPM had functions like that.

**Review by zyga — APPROVED** [review]
The code looks good.

Uh, what a mess. Is openSUSE going closer to RHEL?

File: packaging/opensuse/snapd.spec (hunk 1 above)
Comment (Conan-Kudo) [reply packaging/opensuse/snapd.spec:18]: RPM macro language is a mash of m4 and shell, so it has properties from both.

**Comment by Conan-Kudo** [comment]
@zyga This is partly getting more similar to other RPM distros, and also a consequence of adopting FHS 3.0. Debian and Ubuntu have also done the same thing and started using `/usr/libexec` too.

**Review by Conan-Kudo — APPROVED** [review]

**Comment by bboozzoo** [comment]
Heh, it actually stops at `govendor sync`. Weird, and needs further investigation.

**Review by mvo5 — APPROVED** [review]
Thanks, yes, let's hope they stick with this transition and eventually converge on /usr/libexec :)

**Pull request merged**